server/filestore.go

```diff
@@ -616,6 +616,11 @@ func (mb *msgBlock) rebuildState() (*LostStreamData, error) {
 		index += rl
 	}
 
+	// For empty msg blocks make sure we recover last seq correctly based off of first.
+	if mb.msgs == 0 {
+		mb.last.seq = mb.first.seq - 1
```

Comment:
Do you need to take care of special case with mb.first.seq could be 0, and therefore mb.last.seq would end-up being -1?